fuel-tx/src/transaction/types/input/predicate.rs

```diff
@@ -0,0 +1,38 @@
+use core::ops::{
+    Deref,
+    DerefMut,
+};
+use derivative::Derivative;
+use fuel_types::fmt_truncated_hex;
+
+#[derive(Clone, Default, Derivative)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+#[derive(fuel_types::canonical::Deserialize, fuel_types::canonical::Serialize)]
+#[derivative(Eq, PartialEq, Hash, Debug)]
+pub struct PredicateCode {
+    #[derivative(Debug(format_with = "fmt_truncated_hex::<16>"))]
+    pub bytes: Vec<u8>,
+}
+impl From<Vec<u8>> for PredicateCode {
+    fn from(bytes: Vec<u8>) -> Self {
+        Self { bytes }
+    }
+}
+impl Deref for PredicateCode {
+    type Target = Vec<u8>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.bytes
+    }
+}
+impl DerefMut for PredicateCode {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.bytes
+    }
+}
+
+#[cfg(feature = "da-compression")]
+impl fuel_compression::Compressible for PredicateCode {
+    type Compressed = fuel_compression::RegistryKey;
+}
```

Comment:
We need to implement `CompressibleBy` by using `CompressionContext`

Reply:
I'm trying to figure out why not doing that still allows the current tests to compile.

Reply:
Apparently not needed. Removed `De/CompressionContext` traits  in 9b33143.